File: python/bad_examples/arch_violation/arch_violation.py
"""Deliberate boundary breach: the domain reaching out to adapters.

The layered contract must reject this module when assert.sh copies it into
``src/warehouse/domain/`` for the arch phase to inspect.
"""

from warehouse.adapters.inventory import InMemoryInventoryGateway
from warehouse.domain.quantity import Quantity
from warehouse.domain.sku import Sku

_GATEWAY = InMemoryInventoryGateway()


def reserve_directly(code: str, amount: int) -> object:
    """Domain logic must never touch adapter implementations."""
    return _GATEWAY.reserve(Sku(code), Quantity(amount))

File: typescript/bad_examples/arch_violation/arch_violation.ts
/**
 * Deliberate boundary breach: the domain reaching out to adapters.
 *
 * The layered contract must reject this module when assert.sh copies it into
 * src/domain/ for one depcruise run. NOTE: the import path is written
 * relative to that destination ("../adapters/inventory" from src/domain/),
 * because this file is only ever analyzed from there — never from its
 * fixture location, where every tooling glob excludes it.
 */

import { InMemoryInventoryGateway } from "../adapters/inventory";

export const forbiddenDomainToAdapterEdge = new InMemoryInventoryGateway();
